src/client/builtin_scene/web_content_renderer.cpp

```diff
@@ -1055,16 +1062,10 @@ namespace builtin_scene::web_renderer
       return;
 
     string &text = textComponent->content;
-    auto paragraphStyle = content.paragraphStyle();
-    auto paragraphBuilder = ParagraphBuilder::make(paragraphStyle, fontCollection_);
-    paragraphBuilder->pushStyle(paragraphStyle.getTextStyle());
-    paragraphBuilder->addText(text.c_str(), text.size());
-    paragraphBuilder->pop();
 
-    auto layoutWidth = round(getLayoutWidthForText(content)) + 1.0f;
-    auto paragraph = paragraphBuilder->Build();
-    paragraph->layout(layoutWidth);
-    paragraph->paint(content.canvas(), 0.0f, 0.0f);
```

Comment:
You need to preserve the above painting, and use the painted canvas/bitmap to generate the SDF texture.